kafka_consumer/datadog_checks/kafka_consumer/legacy_0_10_2.py

```diff
@@ -4,45 +4,32 @@
 from __future__ import division
 
 from collections import defaultdict
-from time import time
 
-from kafka import KafkaClient
 from kafka import errors as kafka_errors
 from kafka.protocol.commit import GroupCoordinatorRequest, OffsetFetchRequest
 from kafka.protocol.offset import OffsetRequest, OffsetResetStrategy, OffsetResponse
 from kazoo.client import KazooClient
 from kazoo.exceptions import NoNodeError
 from six import string_types
 
-from datadog_checks.base import AgentCheck, ConfigurationError, is_affirmative
+from datadog_checks.base import ConfigurationError, is_affirmative
 
-from .constants import CONTEXT_UPPER_BOUND, DEFAULT_KAFKA_TIMEOUT, KAFKA_INTERNAL_TOPICS
+from .constants import KAFKA_INTERNAL_TOPICS
 
 
-class LegacyKafkaCheck_0_10_2(AgentCheck):
+class LegacyKafkaCheck_0_10_2(object):
```

Comment:
I'm not convinced this is a better approach for this case; can you explain your rationale for composition over inheritance?

Reply:
- Option 1 is to make LegacyKafkaCheck and NewKafkaCheck both inherit from `KafkaConsumerCheck`. This can't be done as the agent refuses to load `KafkaConsumerCheck` if it's subclassed anywhere.
- Option 2 is to keep a class `KafkaCheck` and override the `__new__` method to return a `LegacyKafkaCheck` instance in case a certain condition is found after connecting to Kafka. This is the current situation and it needs to be removed with the refactor. Indeed we can't retry checking that version condition by setting the check as a `check_initialization`, once the class is initialized we can't change it.
- Option 3 is to make a `KafkaConsumer` main class that inherit from AgentCheck and this is the only class that can inherit from it. We then need a `NewKafkaConsumer` implementation and a `LegacyKafkaCheck` implementation. The common code can now live in `KafkaConsumer`.

With option 3 there are a few design possibilities, I decided to override the `__getattr__` method because of the large number of calls to `self.gauge`, `self.submit_metadata` etc. that are happening in many places.